internal/ignition/ignition.go

```diff
@@ -449,10 +456,15 @@ func sortHosts(hosts []*models.Host) ([]*models.Host, []*models.Host) {
 }
 
 // UploadToS3 uploads the generated files to S3
-func uploadToS3(ctx context.Context, workDir string, clusterID string, s3Client s3wrapper.API, log logrus.FieldLogger) error {
-	for _, fileName := range fileNames {
+func uploadToS3(ctx context.Context, workDir string, cluster *common.Cluster, s3Client s3wrapper.API, log logrus.FieldLogger) error {
```

Comment:
why do you need the cluster if you are using only the ID?

Reply:
I need the hosts.